wallet/src/repository/wallet/mod.rs

```diff
@@ -372,8 +364,24 @@ where
             }
         }
 
+        // Append balance movements of pending blocks
+        state.pending_movements.values().for_each(|x| {
+            keyed_transactions.extend(x.iter().map(|movement| (movement.db_key, movement.clone())));
+        });
+
+        let total_txns = keyed_transactions.len();
+        // Append local pending balance movements (not yet included in blocks)
+        let local_movements = state.local_movements.values().cloned().collect::<Vec<_>>();
```

Comment:
If you are cloning the `movement` later, you don't need to clone it here. You don't even need to collect it because you are using `local_movements` as an iterator:

```suggestion
        let local_movements = state.local_movements.values();
```